catroid/src/main/java/org/catrobat/catroid/ui/BaseExceptionHandler.java

```diff
@@ -23,45 +23,38 @@
 
 package org.catrobat.catroid.ui;
 
-import android.app.Activity;
+import android.content.Context;
 import android.content.SharedPreferences;
 import android.preference.PreferenceManager;
 import android.util.Log;
 
-import com.google.gson.Gson;
-
-import org.catrobat.catroid.BuildConfig;
+import org.catrobat.catroid.utils.CrashReporter;
 
 public class BaseExceptionHandler implements
 		java.lang.Thread.UncaughtExceptionHandler {
 
 	private static final String TAG = BaseExceptionHandler.class.getSimpleName();
 
-	public static final int EXIT_CODE = 10;
-	public static final String RECOVERED_FROM_CRASH = "RECOVERED_FROM_CRASH";
-	public static final String EXCEPTION_FOR_REPORT = "EXCEPTION_FOR_REPORT";
+	private static final int EXIT_CODE = 10;
+	public static final String IS_TESTING_BASE_EXCEPTION_HANDLER = "isTestingBaseExceptionHandler";
 
-	private final SharedPreferences preferences;
+	private SharedPreferences sharedPreferences;
+	private SharedPreferences.Editor editor;
 
-	public BaseExceptionHandler(Activity context) {
-		preferences = PreferenceManager.getDefaultSharedPreferences(context);
+	public BaseExceptionHandler(Context context) {
+		sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
+		editor = sharedPreferences.edit();
 	}
 
 	public void uncaughtException(Thread thread, Throwable exception) {
 		Log.e(TAG, "unhandled exception", exception);
+		CrashReporter.storeUnhandledException(exception);
 
-		SharedPreferences.Editor prefsEditor = preferences.edit();
-		if (BuildConfig.FIREBASE_CRASH_REPORT_ENABLED) {
-			Gson gson = new Gson();
-			String check = preferences.getString(BaseExceptionHandler.EXCEPTION_FOR_REPORT, "");
-			if (check.isEmpty()) {
-				String json = gson.toJson(exception);
-				prefsEditor.putString(EXCEPTION_FOR_REPORT, json);
-				prefsEditor.commit();
-			}
+		if (!sharedPreferences.getBoolean(IS_TESTING_BASE_EXCEPTION_HANDLER, false)) {
+			System.exit(EXIT_CODE);
+			android.os.Process.killProcess(android.os.Process.myPid());
+		} else {
+			editor.putBoolean(IS_TESTING_BASE_EXCEPTION_HANDLER, false);
 		}
```

Comment:
If we choose option one, manually test (or use) the app and the app crashes, the app hangs completely (black screen, no other feedback, not even an App has crashed or restart). I don't like that at all. If the app crashes, the app should restart (or show an App has crashed notification). So option one is off the table.